include/net/nrf_provisioning.h

```diff
@@ -23,98 +33,253 @@ extern "C" {
+ *       provisioning performance and other system operations.
  */
 enum nrf_provisioning_event {
 	/** Provisioning process started. Client will connect to the provisioning service. */
-	NRF_PROVISIONING_EVENT_START,
-	/** Provisioning process stopped. All provisioning commands (if any) executed. */
+	NRF_PROVISIONING_EVENT_START = 0x1,
+
+	/** Provisioning process stopped. All provisioning commands (if any)
+	 *  executed successfully.
+	 */
 	NRF_PROVISIONING_EVENT_STOP,
-	/** Provisioning complete. "Finished" command received from the provisioning service. */
-	NRF_PROVISIONING_EVENT_DONE
+
+	/**
+	 * Provisioning complete. "Finished" command received from the provisioning service.
+	 * Device is now fully provisioned.
+	 */
+	NRF_PROVISIONING_EVENT_DONE,
+
+	/**
+	 * Provisioning process failed, retry recommended. If CONFIG_NRF_PROVISIONING_SCHEDULED
+	 * is enabled, the provisioning will be automatically rescheduled with exponential
+	 * backoff up to a maximum interval.
+	 */
+	NRF_PROVISIONING_EVENT_FAILED,
+
+	/**
+	 * Provisioning process completed, but no commands were received from the server.
+	 * This may indicate the device is already fully provisioned.
+	 */
+	NRF_PROVISIONING_EVENT_NO_COMMANDS,
+
+	/**
+	 * Provisioning process failed due to too many commands for the device to handle.
+	 * Increase CONFIG_NRF_PROVISIONING_CBOR_RECORDS to allow more commands.
+	 * It also might be needed to increase heap size to allocate the larger number of
+	 * commands.
+	 */
+	NRF_PROVISIONING_EVENT_FAILED_TOO_MANY_COMMANDS,
+
+	/**
+	 * Provisioning process failed because the device is not claimed on the
+	 * provisioning service. The device must be claimed using the attestation token
+	 * before provisioning can succeed.
+	 *
+	 * The event carries a pointer to the modem's attestation token in the 'token'
+	 * field and is only provided if CONFIG_NRF_PROVISIONING_PROVIDE_ATTESTATION_TOKEN
+	 * is enabled. The token pointer is valid only during the callback execution and
+	 * should not be stored.
+	 */
+	NRF_PROVISIONING_EVENT_FAILED_DEVICE_NOT_CLAIMED,
+
+	/**
+	 * Provisioning process failed due to incorrect CA certificate configuration.
+	 * Verify the root CA certificate matches the provisioning service.
+	 */
+	NRF_PROVISIONING_EVENT_FAILED_WRONG_ROOT_CA,
+
+	/**
+	 * Provisioning process failed because no valid datetime reference is available.
+	 * Ensure the device has access to network time or SNTP.
+	 */
+	NRF_PROVISIONING_EVENT_FAILED_NO_VALID_DATETIME,
+
+	/**
+	 * Credential handling requires LTE to be deactivated.
+	 * Application should call lte_lc_offline(), conn_mgr_all_if_disconnect(),
+	 * or similar to deactivate LTE connectivity. This event is generated when
+	 * the library needs to write certificates or credentials to the modem,
+	 * which requires the LTE connection to be offline.
+	 *
+	 * Note that conn_mgr_all_if_disconnect() and lte_lc_offline() disconnects
```

Comment:
You can remove lte_lc_offline() from the note since it deactivates both LTE and GNSS.